### PR TITLE
When adding documents, trying to update the primary-key now throw an error

### PR DESCRIPTION
**Pull request by irevoire**
While updating the test suite, I also noticed an issue with the indexed_documents value of failed tasks and had to update it. I also named a bunch of snapshots that had no name, sorry 😬

Fixes https://github.com/meilisearch/meilisearch/issues/3385
Fixes https://github.com/meilisearch/meilisearch/issues/3411

### CODE REVIEW

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Wow, you also added five new tests for this fix. Thank you!
Please, next time fix other stuff in a different commit 🤭

File: index-scheduler/src/batch.rs [hunk 1]
```diff
@@ -985,17 +997,32 @@ impl IndexScheduler {
...
+                        // but to a different value then we can make the whole batch fail.
+                        Some(pk) if primary_key != pk => {
+                            return Err(milli::Error::from(
+                                milli::UserError::PrimaryKeyCannotBeChanged(pk.to_string()),
+                            )
+                            .into());
+                        }
+                        // if the primary key was set and equal to the one already set for
+                        // the index then there is nothing to do.
+                        Some(_) => (),
```
[the remark this answers is not in the document]
Comment (irevoire) [reply index-scheduler/src/batch.rs:1013]: But this is wrong?
You can't make a match non-exhaustive :grimacing:

[the remark this answers is not in the document]
Comment (Kerollmops) [reply index-scheduler/src/batch.rs:1013]: But my suggestion is exhaustive, when you match a `Some`, go inside, see the if and match it or not.

[the remark this answers is not in the document]
Comment (irevoire) [reply index-scheduler/src/batch.rs:1014]: Ah! Yes right! I was a bit lost without the comment

File: index-scheduler/src/autobatcher.rs [hunk 2]
```diff
@@ -173,11 +220,51 @@ impl BatchKind {
...
+            // 2. Else ->
+            //   2.1 If we already have a primary-key ->
+            //     2.1.1 If the task we're trying to accumulate have a pk -> it must be equal to our primary key to continue
+            //     2.1.2 If the task don't have a primary-key -> we can continue
+            //   (We've already ensured that the current batch was correct according to our pk in a previous step of the autobatcher)
+            //   2.2 If we don't have a primary-key ->
+            //     2.2.1 If both the batch and the task have a primary key they should be equal
+            //     2.2.2 If the batch is set to Some(None), the task should be too
+            //     2.2.3 If the batch is set to None -> we can continue
+            //
```
Comment (Kerollmops) [inline index-scheduler/src/autobatcher.rs:234]: We can maybe remove that part as it is already below. I don't like to duplicate applications as we must keep both in think and we will fail to do so.
```suggestion
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Thank you very much. It is much better! I understand that it was harder to do!

File: index-scheduler/src/autobatcher.rs [hunk 3]
```diff
@@ -173,11 +220,51 @@ impl BatchKind {
...
+            // NOTE: We need to negate the whole condition since we're checking if we need to break instead of continue.
+            //       I wrote it this way because it's easier to understand than the other way around.
+            (this, kind) if !(
```
[the remark this answers is not in the document]
Comment (irevoire) [reply index-scheduler/src/autobatcher.rs:237]: Oooof, thanks for catching that!

File: index-scheduler/src/autobatcher.rs (hunk 2 above)
Comment (irevoire) [reply index-scheduler/src/autobatcher.rs:234]: True, it helped me during the implementation, but the in-code comments should be enough. Thanks for the suggestion :+1:

**Review by Kerollmops — APPROVED** [review]
Thank you very much @irevoire! Merging!
bors merge

**Pull request merged**